nixos/modules/services/misc/n8n.nix

```diff
@@ -0,0 +1,433 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.services.n8n;
+
+  baseConfig = builtins.removeAttrs cfg.config [ "listenAddress" "sslKeyFile" "sslCertFile" ] // {
+    listen_address = cfg.config.listenAddress;
+    ssl_key = cfg.config.sslKeyFile;
+    ssl_cert = cfg.config.sslCertFile;
+  };
+  filteredConfig =
+    filterAttrsRecursive (n: v: v != {})
+    (filterAttrsRecursive (n: v: v != {})
+      (filterAttrsRecursive (n: v: v != null && v != {} && v != [] && v != "")
+        baseConfig
+      )
+    );
+  configFile = pkgs.writeText "n8n.conf" (builtins.toJSON filteredConfig);
+in
+{
+  options.services.n8n = {
+
+    enable = mkEnableOption "Enable n8n server";
+
+    openFirewall = mkOption {
+      type = types.bool;
+      default = false;
+      description = "Open ports in the firewall for the n8n web interface.";
+    };
+
+    package = mkOption {
```

Comment:
It looks like the `n8n` package has no build options and only a single version in nixpkgs: what is the value this option provides?

Reply:
just to make it easy to use a custom version ?

Reply:
This is what overrides are for.